app/lib/bk/compat/parsers/circleci/orbs.rb

```diff
@@ -64,6 +61,54 @@ def install_commands(version, install_dir)
           end
         end
 
+        def translate_docker_build(config)
+          [
+            ("# Attaching workspace at #{config['attach-at']}" if config['attach-at']),
+            ('export DOCKER_BUILDKIT=1' if config['use-buildkit']),
+            lint_dockerfile(config),
+            *pull_cache_images(config),
+            generate_build_command(config)
+          ].compact
+        end
+
+        def lint_dockerfile(config)
+          return unless config['lint-dockerfile']
+
+          "# Linting Dockerfile#{' (treating warnings as errors)' if config['treat-warnings-as-errors']}"
+        end
+
+        def pull_cache_images(config)
+          config['cache_from']&.split(',')&.map { |image| "docker pull #{image.strip} || true" } || []
+        end
+
+        def generate_build_command(config)
+          parts = [
+            'docker',
+            ('buildx' if config['cache_to']),
+            'build',
+            '-f', File.join(config['path'] || '.', config['dockerfile'] || 'Dockerfile'),
+            '-t', generate_tag(config),
+            *generate_optional_args(config),
+            config['docker-context'] || '.'
+          ]
+          parts.compact.join(' ')
+        end
+
+        def generate_optional_args(config)
+          [
+            ("--cache-from=#{config['cache_from']}" if config['cache_from']),
+            ("--cache-to=#{config['cache_to']}" if config['cache_to']),
+            ('--load' if config['cache_to']),
+            *config['extra_build_args']&.split&.map(&:strip)
```

Comment:
the nil-safe operators can be avoided with appropriate defaults:

```suggestion
            *config.fetch('extra_build_args', '').split.map(:strip)
```